packages/debug/src/browser/breakpoint/breakpoint-manager.ts

```diff
@@ -0,0 +1,272 @@
+/********************************************************************************
+ * Copyright (C) 2018 Red Hat, Inc. and others.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License v. 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0.
+ *
+ * This Source Code may also be made available under the following Secondary
+ * Licenses when the conditions for such availability set forth in the Eclipse
+ * Public License v. 2.0 are satisfied: GNU General Public License, version 2
+ * with the GNU Classpath Exception which is available at
+ * https://www.gnu.org/software/classpath/license.html.
+ *
+ * SPDX-License-Identifier: EPL-2.0 OR GPL-2.0 WITH Classpath-exception-2.0
+ ********************************************************************************/
+
+import { DebugSession } from "../debug-model";
+import { DebugSessionManager } from "../debug-session";
+import { injectable, inject } from "inversify";
+import { DebugProtocol } from "vscode-debugprotocol";
+import { SourceOpener, DebugUtils } from "../debug-utils";
+import { FrontendApplicationContribution } from "@theia/core/lib/browser";
+import { ActiveLineDecorator, BreakpointDecorator } from "./breakpoint-decorators";
+import { BreakpointStorage } from "./breakpoint-marker";
+import {
+    EditorManager,
+    EditorWidget,
+    Position,
+    TextEditor,
+    MouseTargetType
+} from "@theia/editor/lib/browser";
+import { ExtDebugProtocol } from "../../common/debug-common";
+import { Emitter, Event } from "@theia/core";
+import { BreakpointsApplier } from "./breakpoint-applier";
+
+/**
+ * The breakpoint manager implementation.
+ */
+@injectable()
+export class BreakpointsManager implements FrontendApplicationContribution {
+    protected readonly onDidChangeBreakpointsEmitter = new Emitter<void>();
+
+    constructor(
+        @inject(DebugSessionManager) protected readonly debugSessionManager: DebugSessionManager,
+        @inject(SourceOpener) protected readonly sourceOpener: SourceOpener,
+        @inject(ActiveLineDecorator) protected readonly lineDecorator: ActiveLineDecorator,
+        @inject(BreakpointDecorator) protected readonly breakpointDecorator: BreakpointDecorator,
+        @inject(BreakpointStorage) protected readonly storage: BreakpointStorage,
+        @inject(BreakpointsApplier) protected readonly breakpointApplier: BreakpointsApplier,
```

Comment:
`BreakpointDecorator`, `BreakpointStorage`, so why `BreakpointsApplier` ?

Maybe it would be more consistent to call it `BreakpointApplier` too? (-s)